api/src/main/java/io/smallrye/reactive/messaging/ChannelRegistry.java

```diff
@@ -21,12 +21,16 @@
 
     void register(String name, MutinyEmitter<?> emitter);
 
+    void register(String name, Class<?> emitterType, Object emitter);
```

Comment:
Would `<X> void register(String name, Class<X> emitterType, X emitter);` works?

Reply:
I guess I'd add the type parameter here just for symmetry with `getEmitter` :-)